### ISSUE TITLE
Abolishing Liquidity Incentives And Diverting The Funds To Provide Actual Autonomous Liquidity

### ISSUE THREAD

**Issue by Coinhoarder** [steemit/steem#174]
I [posted](https://steemit.com/sip/@coinhoarder/steem-proposal-abolishing-liquidity-incentives-and-reverting-the-funds-to-provide-actual-autonomous-liquidity) this on Steemit a couple days ago, but it quickly got buried with little to no attention. I thought I would try submitting it here.

# The Problem:

Liquidity incentives are being gamed by liquidity "providers" buying and selling SD back and forth among themselves by using two accounts that they control and a large war chest (a large amount of STEEM or SD). There are many Steemit posts that already cover this problem, so I will not go too far into it. [1]

This results in 1200 STEEM being printed every hour and paid gifted to people who aren't actually providing useful liquidity to STEEM users (which is the intended purpose of liquidity incentives.) At the current market value, STEEM is paying $3,756 USD per hour, $90,144 USD per day, $631,008 per week, $2,208,528 per month or $26,502,336 per year for a broken feature. That money is mostly going to people that are trading among themselves, gaming the intended purpose of liquidity incentives, and not actually providing enough useful liquidity for the amount they are paid.

Obviously liquidity incentives are broken, and there has been word that the developers intend to fix the incentive system somehow (I have not seen any published details.) However I posit that no matter what reward-based incentive system is conceived, or what parameters are tweaked, that someone will eventually figure out a way to game it. Liquidity incentives result in a perpetual "cat and mouse" game, with the system constantly being tweaked, ending up being eventually gamed, then tweaked again (and so on and so forth.)

# The Solution:

Instead of providing game-able incentives for market makers to provide liquidity to the SD market, I propose eliminating the incentives altogether and pivoting to an alternative way to provide liquidity in the SD market.

# Major Benefit

There is one major benefit of my proposal versus any liquidity incentive schemes you could come up with. SP, SD and STEEM stakeholders are not debased with my solution, which is untrue of any liquidity incentive scheme you could come up with that benefits third parties. The STEEM/SD printed for autonomous market making liquidity stays on-chain and on the books for eternity, for the purpose of market making. It never makes its way into the STEEM off ramps as downward pressure on the market. Therefore, even though SD and STEEM are printed for autonomous liquidity operations, SP, SD and STEEM stakeholders are not debased in any meaningful way.

# How It Works:

**Basic Idea:**

I am proposing that we revert the value of 1200 STEEM printed per hour for liquidity incentives to a decentralized and autonomously controlled fund that provides actual and useful liquidity to Steemit users. The fund would autonomously setup buy orders on the SD market an arbitrary percentage away from the median price feed and sell orders an arbitrary percentage above the the median price feed.

**Balancing The Books:**

In certain market conditions, it is plausible that there may be one side of the books whose value heavily outweighs the other. For instance, in a bear market the SD sell side may be much thinner than the buy side, or in a bull market the SD buy side may be much thinner than the sell side. This results in the larger side sitting there mostly useless on the books. There are two ways that the autonomous liquidity fund can offset this. Changing the asset that is printed for liquidity (SD or STEEM), or changing the percentage the liquidity is sold from the median price feed. By using both tools at Steem's disposal, eventually equilibrium will be reached.

Unlike the currently implemented liquidity incentives in which 1200 STEEM is printed a hour, STEEM can autonomously decide whether the same value (1200 worth of STEEM) should be printed in the form of SD or STEEM. The funds printed for the purpose of liquidity should be printed in the form of whichever side of the books is thin. If the SD sell side is thin then the funds should be printed in SD, or if the SD buy side is thin then the funds should be printed in STEEM. Alternatively, both SD and STEEM could be printed proportionately to the weight on each side.

Alternatively, if one side of the books ends up being disproportionate to the other, then the arbitrary percentage the SD/STEEM is being sold away from the median price feed should be reduced on the side that has amassed the greater amount of value. In theory, the closer to the median price feed the SD/STEEM is being sold at, the more likely it is to be sold to third part market makers operating closer to the median price feed than the autonomous liquidity fund.

**Extreme Market Conditions:**

If the books are not automatically balanced by slowly inching the sell orders towards the median price feed, or printing 1200 worth of STEEM in either STEEM or SD, then extreme measures can be utilized to balance the books. Extreme measures such as selling at (or slightly below) the median price feed can be taken if extreme market conditions occur. Selling at the median price feed until the books are balanced may suffice, but I posit the losses from selling slightly lower than the price feed (to even the books in extreme scenarios) would still be much less costly compared to the value that SP/STEEM stakeholders are losing via the currently implemented game-able liquidity incentive program.

**Combating "Whale Gaming":**

If the autonomous liquidity fund is being sold too close to the median price feed, then it is possible for a whale to buy it up and then proceed to sell it at a higher percentage away from the median price feed. If a whale had enough ammunition in their war chest (a large amount of STEEM or SD), then this could prove to be a lucrative opportunity for the whale.

To combat any such "whale gaming", I suggest that the arbitrary percentage the SD/STEEM is sold away from the median price feed start large and slowly creep toward the price feed once an arbitrary (but sufficient) amount of value is built up. For a maximum amount of flexibility, that arbitrary percentage for both sides of the books should operate independently from the other. Eventually, the more SD/STEEM that is built up in the autonomous liquidity fund (it is receiving 1200 STEEM per hour,) the closer the autonomous liquidity sell orders can slowly inch closer toward the peg. The spread will therefore be lessened over time as funds are built up further away from the median price feed.

By selling the initial autonomous liquidity a larger arbitrary percentage away from the peg, this eliminates the profitability of a whale buying it up and selling at a higher spread. For instance, say that arbitrary percentage is 15% away from the median price feed and a whale buys it up. To make any profit from this trade, the whale must then sell the funds at over 15% away from the median price feed. At a certain percentage away from the median price feed, no reasonable person would buy or sell STEEM/SD that far away from the peg as they would be losing too much money in the trade. This leaves the whale, whose intent was to game the autonomous liquidity fund, stuck with SD/STEEM they paid too much for and cannot sale. Their only way out of their position is to sell it at a loss.

Eventually, as the value/amount of SD/STEEM grows at the highest determined arbitrary percentage away from the peg, future STEEM printed for autonomous liquidity purposes can be inched closer and closer to the peg. As the value/amount of both sides of the book builds up, the amount that is on the books will be too much for any whale to profitably "whale game", even as the spread is lessened. Over time as the autonomous liquidity funds build up, the market spread will be less and less.

# Considerations

**Price Feed:**

A recently updated price feed is necessary for this proposal to work. Apparently, delegates are currently able to update the price feed whenever they want, but the system doesn't update the median price feed but once a hour. Someone brought this up to me and claimed that this was a "deal breaker" for the proposal, but I posit that with a 2 second block time that there is no reason why the price feeds couldn't be updated in a sufficient amount of time. Here is how I would propose going about that:

I propose making it a requirement that a delegate had to update their price feed sometime within the previous round to be included in the next round's queue. If they have not updated their price feed within the previous round, then the next delegate with the most votes that has updated their price feed within the past round (and is not currently in the top 19 of voted delegates) takes their place. This ensures that all price feeds have been updated within approximately the past 42 seconds. I think that is sufficiently often enough in order to autonomously maintain a fairly tight peg. Also, there is also no reason why delegates couldn't update faster than every 42 seconds (once every block is feasible if the delegate so chooses).

**References:**

1. Steemit posts covering the issue with liquidity incentives being gamed:

https://steemit.com/steem/@blakemiles84/two-users-are-gaming-the-liquidity-reward-system-and-earning-1200-steem-per-hour

**Comment by jl777** [comment]
I proposed a hybrid solution: https://github.com/steemit/steem/issues/175

**Comment by Coinhoarder** [comment]
I fail to see how that is a hybrid system. I am suggesting doing away with liquidity incentives.

**Comment by Coinhoarder** [comment]
PS: I elaborated (edited) my earlier post jl777, just adding this so GitHub notifies you of it.

**Comment by LeeBlues** [comment]
What about half-Autonomous? It's code that operates fund but group of managers can adjust parameters,variables through internal voting according to market condition. So, multisig fund?

**Comment by Coinhoarder** [comment]
@LeeBlues That could work. Bitshares functions similarly... the top X amount of delegates can decide on and change certain parameters. I forget what they call it, but it could definitely work.

**Comment by abitmore** [comment]
TL;DR Is it possible to manipulate the price on external exchanges to drill the fund in the pool? By simply following prices on other markets, it's likely to lose money.

**Comment by Coinhoarder** [comment]
"TL;DR Is it possible to manipulate the price on external exchanges to drill the fund in the pool? By simply following prices on other markets, it's likely to lose money."

I don't think it would be possible any more so than it is already possible. Ideally, the price feed and buy/sell orders are updated as quickly as possible, so if you manipulate the price on external/internal markets then the buy/sell orders would be changed accordingly. Thanks for bringing this up though, I have thought of two things I need to add and/or clarify in the proposal:

1. I didn't mention how the price feed would be determined in the proposal, so I guess I need to clarify that. If the median price feed was obtained by only tracking SBD/??? trading pairs on centralized exchanges, then I don't think it is game-able assuming it updates at a frequent enough frequency.

2. I think it may also be prudent to have a minimum percentage that the liquidity fund will buy/sell around the median price feed as a "safety net" of sorts in case of market manipulation attempts.

Keep in mind that as you get closer to the median price feed, there will be less and less autonomous liquidity provided. So, even if you are able to somehow game the liquidity fund by manipulating outside markets (which I don't think is possible if it is designed properly), it would not be very lucrative unless you are able to greatly manipulate the market by a large percentage in a very fast time frame. You would need a huge amount of money to do this, as you cannot send to or withdrawal from a centralized exchange fast enough before the autonomous liquidity fund adjusts for changing market conditions. So, you would need a huge amount of money on the centralized exchange as well as on the STEEM chain.

I haven't received much peer review unfortunately, so I can only theorize so much about how well my proposal would or wouldn't work. Admittedly, the economics are quite complicated, and this would need to be peer reviewed more extensively before being put into action.

**Comment by jl777** [comment]
I have a bot running now that is tracking the external exchange prices for both SBD and STEEM and adjusts the internal market bid/asks based on that. It is called "taker" and is doing decently even though i never does any self-trading. of course it only does a fraction of what the self-trading bots do, but it ended up doing much better than I had expected, especially with the 30 minute requirement.

https://steemit.com/money/@jl777/steem-price-might-double-next-week-liquidity-awards-to-be-temporarily-suspended-new-supply-of-liquid-steem-to-be-reduced

**Comment by jl777** [comment]
https://steemit.com/steem/@jl777/a-non-gameable-liquidity-points-system

**Comment by theoreticalbts** [comment]
Closing this ticket due to (1) age and (2) lack of mandate to replace liquidity rewards.
